macros/create_artifact_resources.sql

```diff
@@ -1,29 +1,115 @@
-{% macro create_artifact_resources() %}
+-- NOTE: The `replace` option is designed only to be used during testing.
+-- Using it in production will do a hard reset of your existing artifacts.
+{% macro create_artifact_resources(replace=false) %}
 
 {% set src_dbt_artifacts = source('dbt_artifacts', 'artifacts') %}
+{% set artifact_stage = var('dbt_artifacts_stage', 'dbt_artifacts_stage') %}
+
+{% set src_results = source('dbt_artifacts', 'dbt_run_results') %}
+{% set src_results_nodes = source('dbt_artifacts', 'dbt_run_results_nodes') %}
+{% set src_manifest_nodes = source('dbt_artifacts', 'dbt_run_manifest_nodes') %}
 
 {{ create_schema(src_dbt_artifacts) }}
 
-{% set create_stage_query %}
-create stage if not exists {{ src_dbt_artifacts }}
+{% set create_v1_stage_query %}
+create {% if replace %} or replace {% endif %} stage if not exists {{ src_dbt_artifacts }}
+file_format = (type = json);
+{% endset %}
+
+{% set create_v2_stage_query %}
+create {% if replace %} or replace {% endif %} stage if not exists {{ artifact_stage }}
 file_format = (type = json);
 {% endset %}
 
-{% set create_table_query %}
-create table if not exists {{ src_dbt_artifacts }} (
+{% set create_v1_table_query %}
+create {% if replace %} or replace {% endif %} table if not exists {{ src_dbt_artifacts }} (
```

Comment:
Unfortunately `or replace` and `if not exists` can't be used in the same statement: https://docs.snowflake.com/en/sql-reference/sql/create.html#general-usage-notes

Reply:
For the purposes of these automated integration tests the `replace` doesn't matter as each runs in a separate schema named by the commit hash. Appreciate it's useful for local testing though.

Reply:
ahhhhh - understand. I thought it was _necessary_ for the CI tests! In that case I'm just going to revert the `replace` changes. I think it's almost a _dangerous_ feature if you don't need it.